### ISSUE TITLE
Add support to gradle configuration cache

### ISSUE THREAD

**Issue by apssouza22** [kordamp/jandex-gradle-plugin#28]
It seems that this plugin doesn't support the [gradle configuration](https://docs.gradle.org/current/userguide/configuration_cache.html). Any plan to support it? 

```
- Task `:platform:runtime:quarkus:base:jandex` of type `org.kordamp.gradle.plugin.jandex.tasks.JandexTask`: cannot serialize object of type 'org.gradle.api.internal.artifacts.configurations.DefaultUnlockedConfiguration', a subtype of 'org.gradle.api.artifacts.Configuration', as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.5/userguide/configuration_cache.html#config_cache:requirements:disallowed_types

```

**Comment by aalmiray** [comment]
Is this feature out of incubation?

The docs still list the following:

```

This feature is currently not enabled by default. This feature has the following limitations:

The configuration cache does not support all core Gradle plugins and features. Full support is a work in progress.

Your build and the plugins you depend on might require changes to fulfil the requirements.

IDE imports and syncs do not yet use the configuration cache.
```

**Comment by apssouza22** [comment]
@aalmiray yes. It is in broadly available and with full support for Gradle Java plugin. However it requires some changes in the community plugins to fully support it.

 Quarkus plugin support it and it dramatically improve our build time, however we are blocked to use it with Quarkus  multiple modules project because this plugin.

https://quarkus.io/guides/gradle-tooling#gradle-configuration-cache

**Comment by yk-littlepay** [comment]
Hello @apssouza22 , could you please at least add ` t.notCompatibleWithConfigurationCache("See https://github.com/kordamp/jandex-gradle-plugin/issues/28")` somewhere [here in JandexPlugin](https://github.com/kordamp/jandex-gradle-plugin/blob/6853c81cf03eda3cc92cdf810d48bf30414accf0/src/main/groovy/org/kordamp/gradle/plugin/jandex/JandexPlugin.groovy#L70)

PS. Here is patch to save some time 

[Update_dependencies,_add_notCompatibleWithConfigurationCache.patch](https://github.com/user-attachments/files/18724173/Update_dependencies._add_notCompatibleWithConfigurationCache.patch)

**Comment by aalmiray** [comment]
Before marking the project as non compatible with Gradle CC I'd rather understand what's required to make it happen. In that case a single release could be posted (hopefully shortly) instead of two.

**Comment by re-thc** [comment]
Would be good to actually support CC. There's a [guide](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:requirements) from the official site. Things like `project` can't be referenced directly etc.

**Comment by aalmiray** [comment]
These days I do not spend much time working on Gradle plugin projects. I've been burnout by the constant churn of deprecated idioms and breaking changes. If you've use Gradle since `0.4` as I have you would understand.

This being said, it's possible to sponsor work on this feature via GitHub sponsors, Open Collective, and/or Tidelift.

**Comment by re-thc** [comment]
Definitely thanks for developing this plugin and trying to maintain it. A lot of users come from Quarkus and maybe that's where the problem is. Is this something the Quarkus team can take up instead then?

**Comment by apssouza22** [comment]
Turns out that we don't need to use this plugin with Quarkus. We can use beans.xml instead. https://www.baeldung.com/quarkus-bean-discovery-index#3-framework-specified-implications
